File: PracticaExamen/Logica/Factura.vb
Public Class Factura
    Property ID_Factura As Integer
    Property Fecha As Date
    Property Cliente As Cliente
    Property Detalle As ListaProductos

    Sub New()
        Detalle = New ListaProductos()
    End Sub

    Sub New(id As Integer, f As Date, c As Cliente, d As ListaProductos)
        Me.ID_Factura = id
        Me.Fecha = f
        Me.Cliente = c
        Me.Detalle = d
    End Sub

    Sub New(id As Integer, f As Date, c As Cliente)
        Me.ID_Factura = id
        Me.Fecha = f
        Me.Cliente = c
        Me.Detalle = New ListaProductos()
    End Sub

    Sub New(f As Date, c As Cliente, d As ListaProductos)
        Me.Fecha = f
        Me.Cliente = c
        Me.Detalle = d
    End Sub

    Sub New(f As Date, c As Cliente)
        Me.Fecha = f
        Me.Cliente = c
        Me.Detalle = New ListaProductos()
    End Sub

End Class

File: Practicav2/Logica/Factura.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ObjetosTransferencia;
using AccesoDatos;

namespace Logica
{
    public class Factura
    {
        public int Numero { get; set; }
        public DateTime Fecha { get; set; }
        public Cliente ClienteF { get; set; }
        public DetalleFactura detalle { get; set; }

        public Factura() { }

        public Factura(int num, DateTime fecha, Cliente c, DetalleFactura d) {
            this.Numero = num;
            this.Fecha = fecha;
            this.ClienteF = c;
            this.detalle = d;
        }

        public int guardar() {

            DAOFactura daoF = new DAOFactura();
            TOFactura toF = new TOFactura();
            toF.ClienteF = this.getTOCliente();
            toF.detalle = this.getTODetalle();
            toF.Total = detalle.total();

            toF = daoF.guardar(toF);

            this.Numero = toF.Numero;
            this.Fecha = toF.Fecha;

            return this.Numero;
        }

        private TODetalleFactura getTODetalle() {

            TODetalleFactura toD = new TODetalleFactura();

            foreach (Producto item in detalle.lista)
            {
                toD.lista.Add(new TOProducto(item.Codigo, item.Descripcion, item.Precio));
            }

            return toD;
        }

        private TOCliente getTOCliente() {
            return new TOCliente(ClienteF.Cedula, ClienteF.Nombre);
        }

    }
}
